### ISSUE TITLE
[Feature]: The quotes formatting style should default to none instead of single quotes

### ISSUE THREAD

**Issue by gsf-elefebvre** [interactive-data-language/vscode-idl#25]
### Description

I would make the case that quotes in a language is more associated to the syntax than the actual style and changing them can alter the execution of code. For example if I had a string in IDL and was doing a regex on a json string
```
myStr.regex('"myKey"')
```
If auto format is set to double quotes then the regex behavior changes.
Another example is where IDL invokes another language where interpolation is defined by the quotes such as shell scripting. For example running:
```
spawn, 'echo "$PATH"', result
```
Does not give the same result  as:
```
spawn, "echo '$PATH'", result
```

### Why it Matters

By default running the auto formatter has the potential to create bugs in peoples' code.

### Suggested Behavior

Default the quotes setting in formatter to "none". Create documentation making the user aware of the risks involved when setting it to something else

**Comment by znorman-harris** [comment]
Thanks for the examples! We definitely weren't thinking about some of those cases where the quote style matters.

Let me think about this a bit as I think the  real problem here is flipping quotes and incorrectly changing the interior quotes where they shouldn't change. I think we can adjust the quote formatting to handle this case without too much effort.

As a workaround, you can use backticks for your strings which will preserve the string itself:

```idl
spawn, `echo "$PATH"'`, result
```

**Comment by znorman-harris** [comment]
Alright, this has been fixed! I changed it to *not* flip the quotes inside of strings and preserve the original content.

We now use IDL's quote escape characters (which also get highlighted better after a small fix from one of these examples).

This means that, you'll see this for single -> double quote behaviors:

```idl
; single quote before converting to double
a = 'echo, "myval"'

; single quote after conversion to double
a = "echo, ""myval"""
```

and the following for double -> single quote:

```idl
; double quote before converting to single
a = "echo, 'myval'"

; double quote after conversion to single
a = 'echo, ''myval'''
```

We should have a release out in the next few days, but I can send you a preview build if you watn.
